### ISSUE TITLE
RestResponseCookie is not interchangeable with System.Net.Cookie

### ISSUE THREAD

**Issue by verydarkmagic** [restsharp/RestSharp#1038]
## Expected Behavior
Consider this code
```
var client = new RestClient("https://example.com/api");
var loginRequest = new RestRequest("Login", Method.POST);
var result = client.Execure(loginRequest ); // has a Set-Cookie: SessionID blablabla

//add the session cookie to the client for all further requests
foreach(var cookie in result.Cookies){
        client.CookieContainer.Add(cookie) //<-- I wish this worked
}
```

## Actual Behavior
Error: cannot convert "RestSharp.RestResponseCookie" to "System.Net.Cookie"

Is there a good reason to have a RestResponseCookie (which looks like copied from System.Net.Cookie, all fields are the same) and use it over the standard class? Otherwise this looks like a minor design flaw

**Comment by lslvxy** [comment]
mark+1

**Comment by Planche95** [comment]
Exacly. What is the best way to use Cookies from some restsharp response in other restsharp request?

**Comment by alexeyzimarev** [comment]
Good point. The issue is that if we change it now, it will be a breaking change.

**Comment by alexeyzimarev** [comment]
Found out that we have `HttpCookie` and `RestResponseCookie`, which are completely identical... Hard to say what is the reason for that. I marked one of those obsolete for the start.

**Comment by GieltjE** [comment]
IRestResponse.cs should probably be modified to use the HttpCookie instead of the RestResponseCookie

**Comment by alexeyzimarev** [comment]
As the first step I could propose adding a conversion extension method like `RestResponseCookie.ToHttpCookie`. We can also use it then when creating a new request like `request.AddResponseCookies`. That should be easy to implement.

**Comment by GieltjE** [comment]
Shouldn't be closed or anything as the current situation results in deprecation warnings that can't be fixed.

**Comment by GieltjE** [comment]
Any chance at least the deprecation warning could be fixed sometime?

**Comment by GieltjE** [comment]
!STALE

**Comment by GieltjE** [comment]
!STALE

**Comment by GieltjE** [comment]
!stale && mustfix

**Comment by GieltjE** [comment]
damn bot should be sent to /dev/null

**Comment by GieltjE** [comment]
!stale

**Comment by Moskus** [comment]
VS2019 also remarks that "HttpCookie is obsolete".

**Comment by alexeyzimarev** [comment]
I removed `HttpCookie` entirely in v107. However, cookies were always accessible directly via the cookie container, you should always use one as you'd do with `HttpClient`.

The pre-release package is out there, feel free to test it. If you get any issues with cookies, please open another issue.
